src/chat-handler.js

```diff
@@ -0,0 +1,28 @@
+
+const chatHandler = (route, data) => {
+  switch (route) {
+    case ('CHATROOM.REGISTER') : {
+      newuser = new NewUser(data);
```

Comment:
@njsfield 's point 5 above refers to no. 11 in the mentor journey of the client/server model. So the server-side `chat.js` needs to contain method like `addSocket`, which puts the appropriate socketId into the `endpoints` object.